basic/src/test/java/org/jboss/hal/testsuite/test/runtime/messaging/XaJmsMetricsTestCase.java

```diff
@@ -0,0 +1,161 @@
+            SUBSYSTEM_NAME = "messaging-activemq", CONNECTION_FACTORY_NAME = "activemq-ra", SERVER = "server",
+            DEFAULT = "default", POOLED_CONNECTION_FACTORY = "pooled-connection-factory",
+            STATISTICS_ENABLED = "statistics-enabled";
+    private static final Address defaultJmsServerAddress = Address.subsystem(SUBSYSTEM_NAME).and(SERVER, DEFAULT),
+            queueAddress = defaultJmsServerAddress.and("jms-queue", "test"),
+            pooledConnectionFactoryAddress =
+                defaultJmsServerAddress.and(POOLED_CONNECTION_FACTORY, CONNECTION_FACTORY_NAME),
+            pooledConnectionFactoryStatsAddress;
+    private static ModelNodeResult originalFactoryStatsEnabledResult;
+
+    static {
+        Address subsystemRuntimeAddress = ConfigUtils.isDomain() ?
+                Address.host(getDefaultHost()).and(SERVER, MONITORED_SERVER).and(SUBSYSTEM, SUBSYSTEM_NAME)
+                : Address.subsystem(SUBSYSTEM_NAME);
+        pooledConnectionFactoryStatsAddress = subsystemRuntimeAddress.and(SERVER, DEFAULT)
+                .and(POOLED_CONNECTION_FACTORY, CONNECTION_FACTORY_NAME).and("statistics", "pool");
+    }
+
+    @Drone
+    private WebDriver browser;
+
+    @Page
+    private MessagingStatisticsPage page;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        ModelNode entriesNode = new ModelNodeListBuilder(new ModelNode("java:/jms/queue/test")).build();
+        ops.add(queueAddress, Values.of("entries", entriesNode)).assertSuccess();
+        WebArchive war = ShrinkWrap.create(WebArchive.class, ARCHIVE_NAME + WAR_FILE_EXTENSION);
+        war.addPackage(MessagingStatisticsBean.class.getPackage());
+        deployOps.deploy(war);
+
+        originalFactoryStatsEnabledResult = ops.readAttribute(pooledConnectionFactoryAddress, STATISTICS_ENABLED);
+        originalFactoryStatsEnabledResult.assertSuccess();
+        if (!originalFactoryStatsEnabledResult.booleanValue()) {
+            ops.writeAttribute(pooledConnectionFactoryAddress, STATISTICS_ENABLED, true);
+            adminOps.reloadIfRequired();
+        }
+    }
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        try {
+            deployOps.undeploy(ARCHIVE_NAME + WAR_FILE_EXTENSION);
+            ops.removeIfExists(queueAddress);
+            if (!originalFactoryStatsEnabledResult.booleanValue()) {
+                ops.writeAttribute(pooledConnectionFactoryAddress, STATISTICS_ENABLED,
+                        originalFactoryStatsEnabledResult.value());
+                adminOps.reloadIfRequired();
+            }
+        } finally {
+            IOUtils.closeQuietly(client);
+        }
+    }
+
+    @Test
+    public void jmsXaPoolStatisticsTest() throws Exception {
+        generateStats();
+        page.navigateToDefaultProviderStats().switchToPooledConnectionFactory()
+            .selectConnectionFactory(CONNECTION_FACTORY_NAME);
+        verifyMetrics();
+        generateStats();
```

Comment:
What are the benefits of calling generateStats() again? The statistics were already generated on the first line of this method.